examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -352,81 +394,102 @@ def _mean_frequency_by_risk_group(y_true, y_pred, sample_weight=None,
-
+    y_true = df_test["Frequency"].values
+    exposure = df_test["Exposure"].values
     q, y_true_seg, y_pred_seg = _mean_frequency_by_risk_group(
-        df_test["Frequency"].values,
-        y_pred,
-        sample_weight=df_test["Exposure"].values,
-        n_bins=10)
+        y_true, y_pred, sample_weight=exposure, n_bins=10)
+
+    # Name of the model after the estimator used in the last step of the
+    # pipeline.
+    print(f"Predicted number of claims by {model[-1]}: "
+          f"{np.sum(y_pred * exposure):.1f}")
 
-    axi.plot(q, y_pred_seg, marker='o', linestyle="-", label="predictions")
-    axi.plot(q, y_true_seg, marker='x', linestyle="--", label="observations")
+    axi.plot(q, y_pred_seg, marker='x', linestyle="--", label="predictions")
+    axi.plot(q, y_true_seg, marker='o', linestyle="--", label="observations")
     axi.set_xlim(0, 1.0)
-    axi.set_ylim(0, 0.6)
+    axi.set_ylim(0, 0.5)
     axi.set(
-        title=model[-1].__class__.__name__,
+        title=model[-1],
         xlabel='Fraction of samples sorted by y_pred',
         ylabel='Mean Frequency (y_pred)'
     )
     axi.legend()
 plt.tight_layout()
 
-##############################################################################
-# The ``Ridge`` regression model can predict very low expected frequencies
-# that do not match the data. It can therefore severly under-estimate the risk
-# for some policyholders.
+###############################################################################
+# The dummy regression model predicts a constant frequency. This model is not
+# attribute the same tied rank to all samples but is none-the-less globally
+# well calibrated (to estimate the mean frequency of the entire population).
 #
-# ``PoissonRegressor`` and ``RandomForestRegressor`` show better consistency
-# between predicted and observed targets, especially for low predicted target
-# values.
+# The ``Ridge`` regression model can predict very low expected frequencies that
+# do not match the data. It can therefore severly under-estimate the risk for
+# some policyholders.
 #
-# However, for some business applications, we are not necessarily interested
-# in the ability of the model to predict the expected frequency value, but
-# instead to predict which policyholder groups are the riskiest and which are
-# the safest. In this case, the model evaluation would cast the problem as a
-# ranking problem rather than a regression problem.
+# ``PoissonRegressor`` and ``HistGradientBoostingRegressor`` show better
+# consistency between predicted and observed targets, especially for low
+# predicted target values.
 #
-# To compare the 3 models within this perspective, one can plot the fraction of
+# The sum of all predictions also confirms the calibration issue of the
+# ``Ridge`` model: it under-estimates by more than 3% the total number of
+# claims in the test set while the other three models can approximately recover
+# the total number of claims of the test portfolio.
```

Comment:
It's curious that the HGBDT overestimates the number of claim, much more than the PoissonRegressor, while both curves look pretty similar

Reply:
I am not sure that this difference is significant. One would need to bootstrap to get confidence intervals but this is too expensive and complicated for this example.